### ISSUE TITLE
False Positive `abstract-method` / `W0223` with `NotImplementedError`

### ISSUE THREAD

**Issue by rustygentile** [pylint-dev/pylint#9275]
### Bug description

```python
"""."""

class MyBaseClass:
    """."""

    def thing1(self):
        """."""
        print('Foo...')

    def thing2(self):
        """."""
        raise NotImplementedError

class MyChildClass(MyBaseClass):
    """."""
```

Raising `NotImplementedError` doesn't necessarily imply an abstract method. From the latest Python [documentation](https://docs.python.org/3/library/exceptions.html#NotImplementedError): "...or while the class is being developed to indicate that the real implementation still needs to be added."  

I would suggest removing `NotImplementedError` from the problematic code [examples](https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/abstract-method.html). Or perhaps splitting this into two different messages so if someone wants to use `NotImplementedError` for methods that still need to be added, they can disable just the message that still applies.
```


### Configuration

_No response_

### Command used

```shell
pylint asdf.py
```

### Pylint output

```python
************* Module asdf
asdf.py:14:0: W0223: Method 'thing2' is abstract in class 'MyBaseClass' but is not overridden in child class 'MyChildClass' (abstract-method)
```

### Expected behavior

No errors.

### Pylint version

```shell
pylint 2.17.2
astroid 2.15.5
Python 3.8.16 (default, May 31 2023, 12:44:21)
```

### OS / Environment

N/A

**Comment by jacobtylerwalls** [comment]
Thanks for the idea. I think having two warnings is too niche. There are other ways to tailor your development/testing practices if you find that this message is slowing you down.
